src/ServiceProviders/ProviderManager.php

```diff
@@ -106,6 +110,14 @@ public function credentialManager()
         return $this->credential_manager;
     }
 
+    /**
+     * Give objects an opportunity to initialize themselves from the cli options
```

Comment:
It would be great if users could specify which credentials they would like to attempt the repair on and use the existing values in build metadata for the other credentials.

For example, I may only wish to repair the CI provider API key/token. In the current implementation, I will have to pass all other credentials through the command line so that they are pre-populated and repair is only attempted on the CI provider token.